### PR TITLE
ETL doc including Cloud Store information

### PR DESCRIPTION
**Pull request by Sean-Holcomb**
Cloud Store documentation within a new ETL doc

### CODE REVIEW

**Comment by dwbrown2** [comment]
Should this be a new standalone doc or is there any existing doc(s) it should be integrated with? 

For example, we have this: https://github.com/kubecost/docs/blob/master/diagnostics.md

**Comment by Sean-Holcomb** [comment]
A lot of these detail could be moved into https://github.com/kubecost/docs/blob/master/diagnostics.md, however, it seems like there is still a need for a doc for the ETL and its endpoints

File: ETL.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# ETL
```
Comment (dwbrown2) [inline ETL.md:1]: It feels like this is really about cloud diagnostics and less about general ETL. Would this be the beginning of a broader document?

Also, how do users discover this document? Can we link from Settings/Diagnostics/cloud integration docs?

File: ETL.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+The ETL is a high level structure within Kubecost tasked with, as it name suggest, extracting cost and usage data from various sources (Prometheus, Thanos, Cloud Service Providers), transforming it into Asset and Allocation objects and loading it into caches that allow for fast querying.
+
+## Cloud Stores
+The ETL contains a Map of Cloud Stores, each of which represent an integration with a Cloud Service Provider. Each Cloud Store is responsible for the Cloud Asset and Reconciliation Pipelines which add Out-of-Cluster costs and Adjust Kubecost's estimated cost respectively via cost and usage data pulled from the Cloud Service Provider. Each Cloud Store has a unique identifier called the `ProviderKey` which varies depending on which Cloud Service Provider is being connected to and ensures that duplicate configurations are not introduced into the ETL. 
+
+- AWS: Account Id of the Master Billing Account
```
Comment (dwbrown2) [inline ETL.md:8]: On AWS, it's possible to link just a sub-account... in case that conflicts with ProviderID implementation

File: ETL.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+- GCP: Project Id
+- Azure: Subscription Id
+
+The `ProviderKey` can be used as an argument for the endpoints related to the Cloud Processes (Cloud Assets and Reconciliation), to indicate that the specified operation should only be done on a single Cloud Store rather than all of them, which is the default behaviour. Additionally the Cloud Store keeps track of the Status of the Cloud Connection and Diagnostics for each of the Cloud Processes. The Cloud Connection Status is meant to be used as a tool in determining the health of the Cloud Connection that is the basis of each Cloud Store. The Cloud Connection Status has various failure states that are meant to provide actionable information on how to get your Cloud Connection running properly. These are the Cloud Connection Statuses:
+
+ - INITIAL_STATUS is the zero value of Cloud Connection Status and means that cloud connection is untested. Once
+Cloud Connection Status has been changed and it should not return to this value. This status is assigned on creation
```
Comment (dwbrown2) [inline ETL.md:15]: I think remove _and_ from this line?

File: ETL.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+- AWS: Account Id of the Master Billing Account
+- GCP: Project Id
+- Azure: Subscription Id
+
+The `ProviderKey` can be used as an argument for the endpoints related to the Cloud Processes (Cloud Assets and Reconciliation), to indicate that the specified operation should only be done on a single Cloud Store rather than all of them, which is the default behaviour. Additionally the Cloud Store keeps track of the Status of the Cloud Connection and Diagnostics for each of the Cloud Processes. The Cloud Connection Status is meant to be used as a tool in determining the health of the Cloud Connection that is the basis of each Cloud Store. The Cloud Connection Status has various failure states that are meant to provide actionable information on how to get your Cloud Connection running properly. These are the Cloud Connection Statuses:
```
Comment (dwbrown2) [inline ETL.md:12]: I think I get what Cloud Processes represents, but I would just leary of using too many technical terms that are less relevant to our end users.

File: ETL.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+- Coverage: The window of time that historical subprocess has covered
+- LastRun: The last time that the process ran, updates each time the periodic subprocess runs
+- NextRun: Next scheduled run of the periodic subprocess
+- Progress: Ratio of Coverage to Total amount of time to be covered
+- RefreshRate: The interval that the periodic subprocess runs
+- Resolution: The size of the assets being retrieved
+- StartTime: When the Cloud Process was started
+
+## EndPoints
```
Comment (dwbrown2) [inline ETL.md:48]: How can you just get general diagnostics without taking an action? That seems valuable to me, but maybe I'm mistaken.

**Review by dwbrown2 — COMMENTED** [review]
Awesome stuff! Dropped some comments for you.

File: ETL.md (hunk 2 above)
Comment (Sean-Holcomb) [reply ETL.md:8]: Yeah it is just whatever billing account the cur was, I made this a little more general

File: ETL.md (hunk 4 above)
Comment (Sean-Holcomb) [reply ETL.md:12]: I was trying not to repeat "Cloud Assets and Reconciliation" a lot here, but if "Cloud Processes" is less clear I wont use it.

File: ETL.md (hunk 5 above)
Comment (Sean-Holcomb) [reply ETL.md:48]: Good call, I added the endpoint to the doc

**Comment by dwbrown2** [comment]
@Sean-Holcomb this blocked on me or still working on it? 

Do we know what we want to do for this? https://github.com/kubecost/docs/pull/120#discussion_r701498076 Ease of discovery feels really important to me.

**Comment by Sean-Holcomb** [comment]
@Webb sorry, looks like I didn't save one of the files in my text editor so git didn't see it. Just pushed it up. That section of diagnostics.md is linked from the question mark button of the Cloud sections on the full diagnostics page of the app. I am ready to merge though. Let me know if you have any other feed back

**Comment by dwbrown2** [comment]
All good, feel free to merge when ready unless you want more reviews! 

One other idea, should we link to this resource from these docs?

https://github.com/kubecost/docs/blob/master/aws-cloud-integrations.md#troubleshooting
https://docs.kubecost.com/azure-out-of-cluster.html
https://docs.kubecost.com/gcp-out-of-cluster.html

**Review by dwbrown2 — APPROVED** [review]

**Pull request merged**